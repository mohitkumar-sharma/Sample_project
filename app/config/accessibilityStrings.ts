export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
