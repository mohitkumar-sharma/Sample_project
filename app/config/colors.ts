export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
