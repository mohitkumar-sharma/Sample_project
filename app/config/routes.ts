export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
