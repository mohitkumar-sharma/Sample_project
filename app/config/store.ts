import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
