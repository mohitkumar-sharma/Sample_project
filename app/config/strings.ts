export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
