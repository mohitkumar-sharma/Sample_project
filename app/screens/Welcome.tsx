import React, { useState } from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { UserInfo, DeviceInfo } from '../states';
import config from '../config';
import { INVTextInput, INVButton, INVBanner } from '../components';
import { validateUserName } from '../controls';

const Welcome = (): JSX.Element => {
  /* useDispatch hook from react-redux for dispatching actions*/
  const dispatch = useDispatch();

  /* useSelector hook from react-redux to select device type from redux */
  const deviceType = useSelector(DeviceInfo.selectDeviceType);

  /* local state for storing user name when entered by user */
  const [name, setUserName] = useState('');

  /**
   * This function is responsible for dispatching an action for storing the user name in redux
   */
  const onGoToHomeBtnPress = () => {
    if (validateUserName(name)) {
      dispatch(UserInfo.saveUser({ userName: name }));
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <INVBanner text={`${config.strings.APP_RUNNING_WARNING} ${deviceType}`} />
      <INVTextInput
        containerStyle={styles.textInputContainer}
        placeholder={config.strings.WELCOME_INPUT_PLACEHOLDER}
        placeholderTextColor={config.colors.divider}
        style={styles.textInput}
        onChangeText={(text) => setUserName(text)}
        value={name}
        testID={config.accessibilityStrings.INV_TEXT_INPUT}
      />
      <INVButton
        containerStyle={styles.buttonContainer}
        buttonText={config.strings.GO_TO_HOME}
        onPress={onGoToHomeBtnPress}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: config.colors.appTheme },
  textInputContainer: { marginTop: 145 },
  buttonContainer: { marginTop: 40 },
  textInput: { paddingHorizontal: 5, paddingVertical: 10, color: config.colors.white },
});

export default Welcome;
