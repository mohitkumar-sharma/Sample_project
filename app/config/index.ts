import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
