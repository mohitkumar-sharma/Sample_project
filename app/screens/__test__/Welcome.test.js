import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Welcome from '../Welcome';
import config from '../../config';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';

describe('Welcome', () => {
  const dispatch = jest.fn();
  redux.useDispatch.mockReturnValue(dispatch);

  it('matches snapshot', () => {
    const { toJSON } = render(<Welcome />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('on changing text', () => {
    const textValue = 'Name';
    const { getByTestId } = render(<Welcome />);

    const textInput = getByTestId(config.accessibilityStrings.INV_TEXT_INPUT);
    fireEvent.changeText(textInput, textValue);

    expect(textInput.props.value).toBe(textValue);
  });

  it('on pressing button without text', () => {
    const textValue = '';
    const { getByTestId } = render(<Welcome />);

    const textInput = getByTestId(config.accessibilityStrings.INV_TEXT_INPUT);
    fireEvent.changeText(textInput, textValue);

    const button = getByTestId(config.accessibilityStrings.NORMAL_BUTTON);
    fireEvent.press(button);
  });

  it('on pressing button with text', () => {
    const expectedPayload = [];
    const mockStore = configureStore([]);
    const deviceInfo = {
      version: '1.0',
      deviceName: 'iPHONE X',
      deviceId: 'kjsdka7662581003-tw',
      deviceType: 'Device',
    };
    const store = mockStore({ deviceInfo });

    jest.spyOn(redux, 'useSelector').mockImplementationOnce(() => store.getState().deviceInfo);

    const textValue = 'Name';
    const { getByTestId } = render(<Welcome />);

    const textInput = getByTestId(config.accessibilityStrings.INV_TEXT_INPUT);
    fireEvent.changeText(textInput, textValue);

    const button = getByTestId(config.accessibilityStrings.NORMAL_BUTTON);
    fireEvent.press(button);

    store.dispatch({ type: 'DEVICEINFO', payload: deviceInfo });
    expectedPayload.push({ type: 'DEVICEINFO', payload: deviceInfo });
    expect(store.getActions()).toEqual(expectedPayload);
  });
});
