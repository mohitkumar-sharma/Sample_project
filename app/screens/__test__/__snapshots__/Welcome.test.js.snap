// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1A1924",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFCC00",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 15,
            "fontWeight": "400",
            "margin": 10,
            "padding": 3,
          },
          undefined,
        ]
      }
      testID="banner-text"
    >
      Notice: You are running this app on undefined
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 20,
        },
        Object {
          "marginTop": 145,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter your name here..."
      placeholderTextColor="#E8E8E8"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#FFFFFF",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        }
      }
      testID="inv-text-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E8E8E8",
            "height": 1,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6EB1F7",
        "borderRadius": 8,
        "height": 55,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "marginTop": 40,
        "opacity": 1,
      }
    }
    testID="normal-button"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "400",
            "textAlign": "center",
          },
          undefined,
        ]
      }
      testID="normal-button-text"
    >
      Go To Home
    </Text>
  </View>
</RCTSafeAreaView>
`;
